Assets/Scripts/EyesightSystem/EyeSight.cs

```diff
@@ -0,0 +1,45 @@
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using UnityEditor;
+using UnityEngine;
+using UnityEngine.Events;
+using UnityEngine.PlayerLoop;
+
+[Serializable]
+public class ObjectFoundEvent : UnityEvent<GameObject> {
+}
+
+public class EyeSight : MonoBehaviour {
+    public Vector2 sightDirection;
+    public Vector2 rayOffset;
+
+    [SerializeField] private float rayLength;
+    [SerializeField] private float sightAngle;
+    [SerializeField] private float sightAngleStep;
+    [SerializeField] private LayerMask sightMask;
+    [SerializeField] private LayerMask reportMask;
+    [SerializeField] private ObjectFoundEvent objectFoundEvent;
+
+    void FixedUpdate() {
+        float start = -sightAngle / 2;
+        float end = sightAngle / 2;
+
+        Vector2 startRaycast = new Vector2(transform.position.x + rayOffset.x,
+            transform.position.y + rayOffset.y);
+
+        for (; start <= end; start += sightAngleStep) {
```

Comment:
```suggestion
        for (float rayAngle = -sightAngle / 2f; rayAngle <= sightAngle / 2f; rayAngle += sightAngleStep) {
```

A little nitpick, but I think this will make the code more clear. The reason I think so is because it puts the variable that we change on each iteration into the `for` head **AND** gives it a clear name (instead of `start`). 

Well, you also will have to remove `start` and `end`... And replace all references to `start` inside the `for` loop with `rayAngle`.